SimDataFormats/TrackingHit/interface/PSimHit.h

```diff
@@ -105,6 +107,15 @@ class PSimHit {
    */
   unsigned int trackId() const { return theTrackId; }
 
+  /** In case te SimTrack ID is incremented by the k_tidOffset for hit category definition, this
+   * methods returns the original theTrackId value directly.
+   */
+  unsigned int originalTrackId() const { return (theTrackId > k_tidOffset) ? theTrackId % k_tidOffset : theTrackId; }
```

Comment:
@fabiocos since you are fixing anyhow, I think this could become more simply
```suggestion
  unsigned int originalTrackId() const { return theTrackId % k_tidOffset; }
```

Reply:
@perrotta yes indeed, I have added this in my branch, and I also extend the dumper to test these methods directly...